apps/service_providers/llm_service/state.py

```diff
@@ -32,69 +67,128 @@ def set_chat_metadata(self, key: Chat.MetadataKeys, value):
-    def callback_handler(self) -> BaseCallbackHandler:
-        pass
+    def format_input(self, input: str) -> str:
+        if not self.input_formatter:
+            return input
 
-    @abstractmethod
-    def format_input(self, input: str):
-        pass
+        template = PromptTemplate.from_template(self.input_formatter)
+        context = self.template_context.get_context(template.input_variables)
+        context["input"] = input
+        return template.format(**context)
 
-    @property
-    @abstractmethod
-    def chat(self):
-        pass
+    def get_tools(self):
+        raise NotImplementedError()
+
+    def pre_run_hook(self, input, config, message_metadata):
+        raise NotImplementedError()
 
+    def post_run_hook(self, output, config, message_metadata):
+        raise NotImplementedError()
+
+    def get_chat_model(self):
+        raise NotImplementedError()
+
+    def get_template_context(self, variables: list[str]):
+        raise NotImplementedError()
+
+    def get_prompt(self):
+        raise NotImplementedError()
+
+    def get_chat_history(self, input_messages: list):
+        raise NotImplementedError()
+
+    def check_cancellation(self):
+        raise NotImplementedError()
+
+    def get_assistant_instructions(self):
+        raise NotImplementedError()
+
+    def get_file_type_info(self, attachments: list) -> list:
+        raise NotImplementedError()
+
+    def get_attachments(self, attachment_type: str):
+        raise NotImplementedError()
+
+    def get_input_message_metadata(self, resource_file_mapping: dict[str, list[str]]):
+        raise NotImplementedError()
+
+    def get_output_message_metadata(self, annotation_file_ids: list):
+        raise NotImplementedError()
+
+    def get_messages_to_sync_to_thread(self):
+        raise NotImplementedError()
+
+    def get_metadata(self, key: Chat.MetadataKeys):
+        raise NotImplementedError()
+
+    def get_openai_assistant(self) -> OpenAIAssistantRunnable:
+        raise NotImplementedError()
+
+    def save_message_to_history(
+        self,
+        message: str,
+        type_: ChatMessageType,
+        message_metadata: dict | None = None,
+        experiment_tag: str | None = None,
+    ):
+        raise NotImplementedError()
 
-class ExperimentState(BaseRunnableState):
+
+class ExperimentAdapter(BaseAdapter):
     def __init__(self, experiment: Experiment, session: ExperimentSession, trace_service=None):
         self.experiment = experiment
         self.session = session
         self.trace_service = trace_service or self.experiment.trace_service
         self.template_context = PromptTemplateContext(session, experiment.source_material_id)
+        self.assistant = self.experiment.assistant
 
     @property
-    def chat(self):
-        return self.session.chat
+    def citations_enabled(self):
+        return self.experiment.citations_enabled
+
+    @cached_property
+    def team(self) -> Team:
+        return self.assistant.team
```

Comment:
shouldn't this be `experiment.team`? One thought is that if we use `session.team` then it will work for all the adapter types.